skimage/filters/_lic.py

```diff
@@ -0,0 +1,163 @@
+        See the examples to see which parameter is appropriate each use case.
+    step_size: str, optional
+        Can be either ``'unit_length'`` or ``'unit_time'``, by default
+        ``'unit_length'``. If set to ``'unit_length'``, the integration step is
+        the velocity scaled to unit length. If set to ``'unit_time'``, the step
+        equals the velocity.
+    maximum_velocity: float, optional
+        Is ``None`` by default. If it is not ``None``, the velocity field is
+        mutiplied with a scalar variable s.t. the maximum velocity after
+        multiplication equals ``maximum_velocity``.
+
+    Returns
+    -------
+    line_integral_convolution : ndarray
+        Returned array of same shape as `input`.
+
+    See Also
+    --------
+    scipy.ndimage.filters.convolve1d : Calculate a one-dimensional convolution
+                                       with a kernel.
+    scipy.integrate.ode : Integrate an ordinary differential equation.
+
+    References
+    ----------
+    .. [1] http://dl.acm.org/citation.cfm?id=166151
+
+    Examples
+    --------
+    .. plot:: lic.py
+
+    """
+    from scipy.ndimage.interpolation import map_coordinates
+    input = np.asarray(input)
+    kernel = np.asarray(kernel)
+    velocity = np.asarray(velocity)
+    if (kernel.ndim != 1) or (len(kernel) < 1):
+        raise ValueError('Kernel must be 1-D array of length at least 1')
+    if velocity.shape != (input.shape + (input.ndim,)):
+        raise ValueError('Shape of velocity not compatible with shape of '
+                         'input image')
+    float_type = np.result_type(velocity.dtype, kernel.dtype, input.dtype,
+                                np.float32)
+    if origin is None:
+        center = 0
+    else:
+        center = (len(kernel) // 2) + origin
+    if center < 0:
+        raise ValueError('Origin parameter is too low for the chosen kernel '
+                         'length')
+    if center >= len(kernel):
+        raise ValueError('Origin paremeter is too large for the kernel length')
+    if maximum_velocity is not None:
+        velocity = maximum_velocity * velocity / np.sqrt(np.max(
+            np.sum(np.square(_astype_nocopy(velocity, float_type)), axis=-1)))
+    if weighted not in ('integral', 'average'):
+        raise ValueError("Weighted must be either 'intergral' or 'average'")
+    if step_size not in ('unit_length', 'unit_time'):
+        raise ValueError("Step_size must be either 'unit_length' "
+                         "or 'unit_time'")
+    if weighted == 'average':
+        weight = np.zeros(input.shape, float_type)
+    result = np.zeros(input.shape, float_type)
+    # the kernel is divided into a part left of "center" and a part right of
+    # "center".
+    # The part on the "right" lies in positive flow direction (sign=1),
+    # the part on the "left" lies in negative flow direction (sign=-1)
+    for sign, ids in [(+1, range(center, len(kernel))),
```

Comment:
Instead of using `map_coordinates`, would `skimage.transform.warp` simplify things?